### PR TITLE
Macro, interface and some help methods with unit testing

### PR DESCRIPTION
**Pull request by TheresePetersson**
*` suspected(), reset(), `call functions to resolver() and a help method (line 15-17 in run-method) with unit tests

### CODE REVIEW

File: modules/primary_monitoring/failure_detector.py [hunk 1]
```diff
@@ -1,33 +1,88 @@
...
+    def get_pend_reqs(self):
+        """Calls get_pend_reqs in Replication module"""
+        return self.resolver.execute(Module.REPLICATION_MODULE,
+                                     Function.GET_PEND_REQS)
+
+    # Added functions
+    def check_progress_by_prim(self, prim):
+        """Checks for progress done by the primary.
+
+        Line 15-17
```
Comment (axelniklasson) [inline modules/primary_monitoring/failure_detector.py]: Nit: Lines 17-19 in the new version of the algo

File: tests/unit_tests/test_failure_detector.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from unittest.mock import Mock, MagicMock
+from resolve.resolver import Resolver
+from resolve.enums import Function, Module
+from modules.replication.models.client_request import ClientRequest
+from modules.replication.models.operation import Operation
+from modules.enums import OperationEnums
+from modules.primary_monitoring.failure_detector import FailureDetectorModule
+
+class TestConfig(unittest.TestCase):
```
Comment (axelniklasson) [inline tests/unit_tests/test_failure_detector.py]: Nit: Should be `TestFailureDetector`

File: tests/unit_tests/test_failure_detector.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    def setUp(self):
+        self.resolver = Resolver()
+        self.clientRequest1 = ClientRequest(1, 1, Operation(OperationEnums.APPEND, 1))
+        self.clientRequest2 = ClientRequest(2, 2, Operation(OperationEnums.APPEND, 1))
```
Comment (axelniklasson) [inline tests/unit_tests/test_failure_detector.py]: Nit: Might be nice to have `clientRequest2` be `APPEND 2` instead to be able to tell them apart

**Review by axelniklasson — APPROVED** [review]
Awesome, only nits

File: tests/unit_tests/test_failure_detector.py (hunk 3 above)
Comment (TheresePetersson) [reply tests/unit_tests/test_failure_detector.py]: (Y)

**Pull request merged**